src/molecules/VoiceRecorder/index.tsx

```diff
@@ -0,0 +1,303 @@
+  const DIALOG_MAX_LENGTH: number = 60 * 1000
+  let IS_RECORDING: boolean = false
+
+  const startRecording = async () => {
+    IS_RECORDING = true
+    record()
+  }
+
+  const stopRecording = () => {
+    IS_RECORDING = false
+    if (mediaRecorder !== null) {
+      mediaRecorder.stop()
+      setMediaRecorder(null)
+    }
+  }
+
+  function record() {
+    navigator.mediaDevices.getUserMedia({ audio: true }).then((stream) => {
+      //start recording:
+      const recorder = new MediaRecorder(stream)
+      recorder.start()
+      setMediaRecorder(recorder)
+
+      //save audio chunks:
+      const audioChunks: BlobPart[] = []
+      recorder.addEventListener('dataavailable', (event) => {
+        audioChunks.push(event.data)
+      })
+
+      //analysis:
+      const audioContext = new AudioContext()
+      const audioStreamSource = audioContext.createMediaStreamSource(stream)
+      const analyser = audioContext.createAnalyser()
+      analyser.minDecibels = VOICE_MIN_DECIBELS
+      audioStreamSource.connect(analyser)
+      const bufferLength = analyser.frequencyBinCount
+      const domainData = new Uint8Array(bufferLength)
+
+      //loop:
+      let time: Date = new Date()
+      let startTime: number
+      let lastDetectedTime: number = time.getTime()
+      let anySoundDetected: boolean = false
+      const detectSound = () => {
+        //recording stopped by user:
+        if (!IS_RECORDING) return
+
+        time = new Date()
+        const currentTime = time.getTime()
+
+        //time out:
+        if (currentTime > startTime + DIALOG_MAX_LENGTH) {
+          recorder.stop()
+          return
+        }
+
+        //a dialog detected:
+        if (
+          anySoundDetected === true &&
+          currentTime > lastDetectedTime + delayBetweenDialogs
+        ) {
+          recorder.stop()
+          return
+        }
+
+        //check for detection:
+        analyser.getByteFrequencyData(domainData)
+        for (let i = 0; i < bufferLength; i++)
+          if (domainData[i] > 0) {
+            anySoundDetected = true
+            time = new Date()
+            lastDetectedTime = time.getTime()
+          }
+
+        //continue the loop:
+        window.requestAnimationFrame(detectSound)
+      }
+      window.requestAnimationFrame(detectSound)
+
+      //stop event:
+      recorder.addEventListener('stop', () => {
+        //stop all the tracks:
+        stream.getTracks().forEach((track) => track.stop())
+        if (!anySoundDetected) return
+
+        //send to server:
+        const audioBlob = new Blob(audioChunks, { type: 'audio/mp3' })
+        makeComputeAPICall(audioBlob)
+      })
+    })
+  }
+  const makeComputeAPICall = async (blob: Blob) => {
+    try {
+      setApiCallStatus('processing')
+      console.log('base', blob)
+      toast.success(`${config.component.messageRecorderWait}`)
+
+      // Define the API endpoint
+      const apiEndpoint = process.env.REACT_APP_BASE_URL
```

Comment:
Remove implementation of this function after the initial toast, we leave it up to who uses this molecule to implement this function on his own and mention this fact in documentation.